core/trino-spi/src/main/java/io/trino/spi/block/Block.java

```diff
@@ -227,7 +231,7 @@ default long getLogicalSizeInBytes()
      * {@code consumer} should be called at least once with the current block and
      * must include the instance size of the current block
      */
-    void retainedBytesForEachPart(BiConsumer<Object, Long> consumer);
+    void retainedBytesForEachPart(RetainedBytesPerPartConsumer consumer);
```

Comment:
We could use `ObjLongConsumer<Object>` here

Reply:
Fair point, that would also be an option and I can make the change if you'd prefer it. Usages are fairly specific and localized to `PageProcessor` and `BlockBigArray` and defined as a lambda which could resolve to any interface type the method signature requires (except for case in a unit test).